pkg/prom/wal/wal.go

```diff
@@ -622,9 +631,23 @@ func (a *appender) AddFast(ref uint64, t int64, v float64) error {
 	return nil
 }
 
-func (a *appender) AppendExemplar(ref uint64, l labels.Labels, e exemplar.Exemplar) (uint64, error) {
-	// remote_write doesn't support exemplars yet, so do nothing here.
-	return 0, nil
+func (a *appender) AppendExemplar(ref uint64, _ labels.Labels, e exemplar.Exemplar) (uint64, error) {
+	s := a.w.series.getByID(ref)
+	if s == nil {
+		return 0, fmt.Errorf("unknown series ref. when trying to add exemplar: %d", ref)
```

Comment:
Can you double check that this will never be true for the first exemplar for any series? i.e., when the first sample for a new series comes in, ref is always 0. I'd like to make sure that's not true for AppendExemplar.

Reply:
Yes, the current tsdb [implementation](https://github.com/prometheus/prometheus/blob/ef584a9df6b4d5b360aa99ddee605b9b92f5ad45/tsdb/head.go#L1385-L1388) for AppendExemplar doesn't create the series entry.  The series will already exist because of the way exemplars are scraped today:   on the same line as a sample, and the sample is processed first ([ref](https://github.com/prometheus/prometheus/blob/ef584a9df6b4d5b360aa99ddee605b9b92f5ad45/scrape/scrape.go#L1422-L1450)).